### PR TITLE
SONAR-9126 Make it possible to prevent the processing of an analysis report by Compute Engine

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java [hunk 1]
```diff
@@ -146,6 +152,14 @@ private void checkOrganizationKeyConsistency(ScannerReport.Metadata reportMetada
     }
   }
 
+  private void checkOrganizationCanExecuteAnalysis(Organization organization) {
+    try {
+      subscriptionBarriers.checkOnProjectAnalysis(new SubscriptionBarrier.Organization(organization.getKey(), organization.getUuid()));
+    } catch (SubscriptionBarrierException e) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: server/sonar-server/src/test/java/org/sonar/server/organization/OrganizationActionValidatorProxyTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.organization;
+
+public class OrganizationActionValidatorProxyTest {
```
Comment (SonarTech) [inline server/sonar-server/src/test/java/org/sonar/server/organization/OrganizationActionValidatorProxyTest.java]: BLOCKER Add some tests to this class. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2187)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java [hunk 3]
```diff
@@ -146,6 +152,14 @@ private void checkOrganizationKeyConsistency(ScannerReport.Metadata reportMetada
     }
   }
 
+  private void checkOrganizationCanExecuteAnalysis(Organization organization) {
+    try {
+      billingValidations.checkOnProjectAnalysis(new BillingValidations.Organization(organization.getKey(), organization.getUuid()));
+    } catch (BillingValidationsException e) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/step/LoadReportAnalysisMetadataHolderStep.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidations.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      super(message);
+    }
+
+    /**
+     * Does not fill in the stack trace
+     *
+     * @see java.lang.Throwable#fillInStackTrace()
+     */
+    @Override
+    public synchronized Throwable fillInStackTrace() {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidations.java]: why do you overload this method?

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsExtension.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.sonar.server.organization;
+
+import org.sonar.api.ce.ComputeEngineSide;
+import org.sonar.api.server.ServerSide;
+
+@ServerSide
+@ComputeEngineSide
+public interface BillingValidationsExtension extends BillingValidations {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsExtension.java]: add `// marker interface` and its role

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsProxy.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.organization;
+
+public interface BillingValidationsProxy extends BillingValidations {
+
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsProxy.java]: add `// marker interface` and its role

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidations.java (hunk 4 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidations.java]: Oh sorry, I was thinking on another stuff, the goal of this method is to not generate a stacktrace as this is functional error

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsExtension.java (hunk 5 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsExtension.java]: ?

File: server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsProxy.java (hunk 6 above)
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/organization/BillingValidationsProxy.java]: ?

**Pull request merged**